### PR TITLE
DT-6461: Use geolocation to improve accuracy of navigation instructions

### PR DESCRIPTION
**Pull request by Antiik91**
## Proposed Changes

  - Position from position store, if available. 
  - Show wait at stop if position is near a stop for 30 seconds.
  - Better handling of stack messages
  - Info message support. 

## Pull Request Check List

  - A reasonable set of unit tests is included
  - Console does not show new warnings/errors
  - Changes are documented or they are self explanatory
  - This pull request does not have any merge conflicts
  - All existing tests pass in CI build

## Review

  - Read and verify the code changes
  - Test the functionality by running the UI locally with all popular browsers available in your platform
  - Check that the implementation matches the design, when such one is defined in a Jira issue
  - Merge the pull request

### CODE REVIEW

File: app/component/itinerary/NaviContainer.js [hunk 1]
```diff
@@ -23,7 +23,23 @@ const legQuery = graphql`
...
+          name
+          code
+          platformCode
+          vehicleMode
+          zoneId
+          parentStation {
+            name
+          }
+        }
+      }
```
Comment (vesameskanen) [inline app/component/itinerary/NaviContainer.js]: I don't think it makes sense to query static information every 10 s.

Query only things that may change. Deeply nested  changing fields should be handled in code line 106 object merge.

File: app/component/itinerary/NaviLeg.js [hunk 2]
```diff
@@ -1,27 +1,28 @@
...
 
 const iconMap = {
   BICYCLE: 'icon-icon_cyclist',
   CAR: 'icon-icon_car-withoutBox',
   SCOOTER: 'icon-icon_scooter_rider',
   WALK: 'icon-icon_walk',
+  WAIT: 'icon-icon_navigation_wait',
 };
 
 /* eslint-disable jsx-a11y/no-static-element-interactions, jsx-a11y/click-events-have-key-events */
```
Comment (vesameskanen) [inline app/component/itinerary/NaviLeg.js]: Lint disable not needed, remove

File: app/component/itinerary/NaviLeg.js [hunk 3]
```diff
@@ -30,12 +31,12 @@ export default function NaviLeg({ leg, nextLeg }, { intl }) {
...
-          <NaviDestination leg={leg} />
+          <NaviLegContent
+            leg={leg}
+            nextLeg={nextLeg}
+            instructions={instructions}
+            legType={legType}
+          />
         </div>
       </div>
     </div>
```
Comment (vesameskanen) [inline app/component/itinerary/NaviLeg.js:39]: Remove unnecessary top div layer

File: app/component/itinerary/NaviLegContent.js [hunk 4]
```diff
@@ -0,0 +1,71 @@
...
+import NaviDestination from './NaviDestination';
+import RouteNumber from '../RouteNumber';
+
+export default function NaviLegContent(
+  { leg, nextLeg, instructions, legType },
+  { intl },
+) {
+  if (legType === 'move') {
+    return (
+      <>
```
Comment (vesameskanen) [inline app/component/itinerary/NaviLegContent.js:15]: remove unnecessary <>

File: app/component/itinerary/NaviStack.js [hunk 5]
```diff
@@ -22,12 +22,8 @@ const NaviStack = ({ notifications, handleRemove, show }) => {
 };
 
 NaviStack.propTypes = {
-  notifications: PropTypes.arrayOf(
-    PropTypes.shape({
-      id: PropTypes.string.isRequired,
-      severity: PropTypes.string.isRequired,
-    }),
-  ).isRequired,
+  // eslint-disable-next-line
```
Comment (vesameskanen) [inline app/component/itinerary/NaviStack.js:25]: Why did you remove proper typing?

File: app/component/itinerary/NaviTop.js [hunk 6]
```diff
@@ -1,35 +1,48 @@
...
-  const [show, setShow] = useState(true);
+  const [showMessages, setShowMessages] = useState(true);
   // All notifications including those user has dismissed.
-  const [notifications, setNotifications] = useState([]);
+  const [messages, setMessages] = useState(new Map());
   // notifications that are shown to the user.
-  const [activeNotifications, setActiveNotifications] = useState([]);
+  const [activeMessages, setActiveMessages] = useState([]);
   const focusRef = useRef(false);
+  // Destination ounter. How long user has been at the destination. * 10 seconds
```
Comment (vesameskanen) [inline app/component/itinerary/NaviTop.js]: typo, counter

File: app/component/itinerary/NaviTop.js [hunk 7]
```diff
@@ -1,35 +1,48 @@
...
 import Icon from '../Icon';
 import NaviStack from './NaviStack';
-import { getAlerts, getScheduleInfo } from './NaviUtils';
+import {
+  getItineraryAlerts,
+  getTransitLegState,
+  getAdditionalMessages,
+} from './NaviUtils';
+
+const DISTANCE_FROM_DESTINATION = 20; // meters
```
Comment (vesameskanen) [inline app/component/itinerary/NaviTop.js]: Maybe  DESTINATION_RADIUS ?

**Review by vesameskanen — CHANGES_REQUESTED** [review]
View crashes quite often like this:

Cannot read properties of null (reading 'time')
TypeError: Cannot read properties of null (reading 'time')
    at getTransitLegState (webpack:///./app/component/itinerary/NaviUtils.js?:174:89)

File: app/component/itinerary/NaviLegContent.js (hunk 4 above)
Comment (Antiik91) [reply app/component/itinerary/NaviLegContent.js:15]: JSX expressions must have one parent element. I'm confused by what you mean by this?

Comment (vesameskanen) [reply app/component/itinerary/NaviLegContent.js:15]: you are right

File: app/component/itinerary/NaviUtils.js [hunk 8]
```diff
@@ -49,23 +72,13 @@ export const getScheduleInfo = (leg, intl) => {
...
-      <div className="navi-info-content">
-        <FormattedMessage
-          id="navileg-mode-citybike"
-          values={{ available: bikes }}
-        />
-      </div>
-    );
-    severity = 'INFO';
-  } else if (late) {
+  const isRealTime = realtimeState && realtimeState === 'UPDATED';
```
Comment (vesameskanen) [inline app/component/itinerary/NaviUtils.js]: testing realtimeState first makes no sense, latter test is enough

**Review by vesameskanen — COMMENTED** [review]
It seems that new incoming messahges are not compared against all messages. Closed messages will reappear.

**Review by vesameskanen — APPROVED** [review]
